PriceCalculatorsKataTests/ProductTest.cs

```diff
@@ -32,20 +36,52 @@ public void productPricePercision()
         [Test]
         public void productTaxAndDiscountCalculation()
         {
-            var productWithDefaultTax = new Product(0)
+            Product productWithDefaultTax = new Product(1,20.25f)
             {
-                Name = "testing product",
-                BasePrice = 20.25f,
+                Name = "s",
             };
-
-
-
-            Product.discount = 15;
-
             Assert.AreEqual(21.26f, productWithDefaultTax.FinalPrice);
+        }
 
-
+        [Test]
+        public void ProductWithSpecialDiscountFinalPriceCalculations1()
+        {
+            UPCBasedDiscount.UPCDiscounts.Add(12345,7);
+            Product product = new Product(12345, 20.25f)
+            {
+                Name = "The Little Prince"
+            };
+            Assert.AreEqual(19.84f,Math.Round(product.FinalPrice),2);
         }
+        
+        [Test]
+        public void ProductWithSpecialDiscountFinalPriceCalculations2()
+        {
+            UniversalTax.Tax = 21;
+            
+            UPCBasedDiscount.UPCDiscounts.Add(789,7);
+            Product product = new Product(12345, 20.25f)
+            {
+                Name = "The Little Prince"
+            };
+            Assert.AreEqual(21.46f,Math.Round(product.FinalPrice),2);
+        }
+
+        [Test]
+        public void ProductReporting()
+        {
+            UPCBasedDiscount.UPCDiscounts.Add(12345,7);
+            Product product = new Product(12345, 20.25f)
+            {
+                Name = "The Little Prince",
+            };
+            product.Report(s =>
+            {
+                Regex rx = new Regex("(?<=(discount).*)([0-9]+.[0-9]+)", RegexOptions.IgnoreCase);
```

Comment:
it's better to have simpler asserts, 
you can just construct the expected text and pass it to the assertion